### PR TITLE
Task 18/logging

### PR DESCRIPTION
**Pull request by rudyeila**
Added some basic logging instead of the print statements. Please take a look and try to test it with postman (didn't do that myself).
Also renames the db/config.py file to db_config since I created a global config.py at the server level and didn't want namespace issues to occur. 

In the future, when creating a new module, the logger setup can be simply copied from another module and used in the same way.
@NaseemSrour In db_item.py you were returning the error messages. I left that behaviour in there but I added error logging ontop of it. Please consider removing the return statements as that could lead to runtime errors (since the functions are expected to return a DB Item)

### CODE REVIEW

File: server/database/db_controller.py [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+import logging 
+
 import mysql.connector
 from mysql.connector import errorcode
 import json
-import config
+import db_config
+from config import global_log_level
+
+logging.basicConfig(level=global_log_level)
```
Comment (NaseemSrour) [inline server/database/db_controller.py:13]: Maybe add a parameter `encoding='utf-8'`

**Review by NaseemSrour — COMMENTED** [review]
Nicely done!
I've read how easy it is to use the logger in other modules.
For the future I would suggest that each module would log into its own logfile (in its folder), thus if we want to see past logs for specific module, we would have the file ready. Alongside a logfile for the whole server side.
As for now we want to see the logs in the console as they are currently configured, because it's easier while developing to see instant logs of your actions.

**Comment by NaseemSrour** [comment]
I'll replace the return statements in another PR after merging this one.

**Comment by NaseemSrour** [comment]
Logger now seems more configurable, which is good for us I believe.
I suggest testing the logger before pushing it.

**Comment by rudyeila** [comment]
Yep I am not done yet. Just finished working for today. Will comment here when its ready for review again.

**Comment by rudyeila** [comment]
I changed the configuration now, so that per package, a log file is created, where all the logging from that package will be forwarded. On top of that, the logging is forwarded to the console. 

All of the package specific logs are propogated to the root logger, which doesn't show anything on the console (in order to avoid duplicates), but saves everything in log_root.log.. That means that file contains all of the logging from the app. 

Testing this is a bit difficult now since I am not sure how to trigger the different functionalities. What I did was simply add logging statement at the Item.get() method and I tested it by making a get request. The logging then behaved as expected it. Beyond that, we'll see once we have more functionality.

**Review by NaseemSrour — COMMENTED** [review]
Looks good. I can test it when I write the test for DBItem. Which is my next task.
We'll see how the logs behave when our test uses DBItem's funtionality.

**Pull request merged**